### DISCUSSION TITLE
Loading a previous trained model into a new dataset

### DISCUSSION THREAD

**Discussion by heatherkristine** [talmolab/sleap#1598] [Help!] [answered]
Would someone be willing to explain the basics on uploading a trained model into a new SLEAP dataset. Or perhaps there is a good resource available explaining this?

**Comment by amblypatty** [comment, marked as answer]
Hi @heatherkristine,

For clarification, are you saying that you have a model folder that you trained using SLEAP, and now you want to perform inference and tracking on a new dataset using that trained SLEAP model? If so, there are a few ways to do so which depend on which way you are using SLEAP:

If you are using the GUI (graphical user interface):

- Utilize the [Training and Inference](https://sleap.ai/tutorials/initial-training.html) tutorial on the sleap.ai website.
- You may either add new videos to the current project that you trained on and go to "Predict" -> "Run Inference...". Then you can select the current video (that you just uploaded to the project) or all videos in the project to predict on from the "Predict On" dropdown list. In the "Model Configuration Tab", you can select the training_config.json file that was used to train your model from your trained model's folder.
- Additionally, you may also create a new SLEAP Project and load the new video dataset there. You'll also have to load the skeleton from the previous project. Then follow the same steps as above.

If you are using the CLI (command line interface):

- Utilize the commands found in the [CLI Guides](https://sleap.ai/guides/cli.html) from the sleap.ai website to call `sleap-track` which is the main command that performs inference on datasets. 
- When calling `sleap-track` you will use the `-m MODELS` command followed by the `--video.dataset VIDEO.DATASET` (replace the capitalized words with the file path to the model folder and the new video file path respectively).

If you are using the python API (application programming interface):

- Utilize the [Interactive and Realtime Inference](https://sleap.ai/notebooks/Interactive_and_realtime_inference.html) notebook from the sleap.ai website.
- You will load your trained model with the `predictor = sleap.load_model()` function and the new video dataset with the `video = sleap.load_video()` function. The notebook guides you through turning your video into a numpy array and then using the model as the "predictor" to predict poses on the numpy array using the `predictor.predict()` function.

I hope this helps clarify the very basics of performing inference from a trained model on new data. It is important to ensure that the new data is similar to the data that was used to train the model, otherwise you may get poor inference results.

Best of luck SLEAPing!

**Reply by talmo** [reply]
Perfect answer!! Thanks @amblypatty!!

@heatherkristine feel free to let us know if you have any other questions or run into any issues!

Cheers,

Talmo
